### ISSUE TITLE
Non-commit object IDs cause panic in raw content API

### ISSUE THREAD

**Issue by gdewald** [gogits/gogs#4855]
- Commit ref: af3911b83319609bff5f582a2e904027fffdce3a
- Git version: 2.11.3
- Operating system: alpine:3.5 (docker)
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [x] Yes: `https://try.gogs.io/api/v1/repos/gdewald/blobtest/raw/1bfbeccdf304c4c98c751e1eed40ef7cb129e17e/Foo`
  - [ ] No
  - [ ] Not relevant
- Log gist:

```
[Macaron] 2017-11-04 22:27:35: Started GET /api/v1/repos/admin123/test/raw/1bfbeccdf304c4c98c751e1eed40ef7cb129e17e/Foo for 192.168.22.1
[Macaron] PANIC: runtime error: slice bounds out of range
/usr/local/go/src/runtime/panic.go:489 (0x43348f)
/usr/local/go/src/runtime/panic.go:35 (0x43217e)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/gogits/git-module/repo_commit.go:61 (0x83a599)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/gogits/git-module/repo_commit.go:115 (0x83aac8)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/gogits/git-module/repo_commit.go:143 (0x83ad2b)
/tmp/go/src/github.com/gogits/gogs/pkg/context/repo.go:358 (0xbff97b)
/usr/local/go/src/runtime/asm_amd64.s:514 (0x45f9e8)
/usr/local/go/src/reflect/value.go:434 (0x4b1b7f)
/usr/local/go/src/reflect/value.go:302 (0x4b1144)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:177 (0x87203f)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:137 (0x871a0a)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x89fb72)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:112 (0x89fa96)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/session/session.go:186 (0x8f7f50)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:79 (0x89f911)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:157 (0x871d12)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:135 (0x871b0b)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x89fb72)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:112 (0x89fa96)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/recovery.go:161 (0x8b33fb)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/logger.go:40 (0x8a3028)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:157 (0x871d12)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:135 (0x871b0b)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x89fb72)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:112 (0x89fa96)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/logger.go:52 (0x8b2674)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/logger.go:40 (0x8a3028)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:157 (0x871d12)
/tmp/go/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:135 (0x871b0b)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x89fb72)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/router.go:187 (0x8b46c9)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/router.go:296 (0x8ac76d)
/tmp/go/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/macaron.go:220 (0x8a466c)
/usr/local/go/src/net/http/server.go:2568 (0x6dbad2)
/usr/local/go/src/net/http/server.go:1825 (0x6d7de2)
/usr/local/go/src/runtime/asm_amd64.s:2197 (0x4621a1)
[Macaron] 2017-11-04 22:27:36: Completed GET /api/v1/repos/admin123/test/raw/1bfbeccdf304c4c98c751e1eed40ef7cb129e17e/Foo 500 Internal Server Error in 224.985593ms
```

## Description
Passing a valid non-commit object ID (e.g. blob ID) as the ref parameter to the [Download raw content](https://github.com/gogits/go-gogs-client/wiki/Repositories-Contents#download-raw-content) API results in HTTP 500.

The issue root cause is as follows:
As is used in [Repository#getCommit](https://github.com/gogits/git-module/blob/master/repo_commit.go#L100), `git cat-file -p <object ID>` returns object content for any type of object without returning an error.

I've fixed it in my fork [here](https://github.com/gdewald/git-module/commit/8816d255a35ce8ed4398859e0250e1592df71613), will submit a pull request.

**Comment by gdewald** [comment]
Pull request posted: https://github.com/gogits/git-module/pull/34

**Comment by Unknwon** [comment]
This is claimed to be fixed by merging gogits/git-module#34, please help test on `develop` branch or https://try.gogs.io.
